### ISSUE TITLE
Decrypted m.js

### ISSUE THREAD

**Issue by cron13** [cazala/coin-hive-stratum#100]
hey guys, can u release decrypted version of m.js, I want to recompile it. At the moment many antiviruses detect it as a virus

**Comment by cazala** [comment]
https://gist.github.com/cazala/50ab1e141736c8a327b26b923a1448e6

**Comment by cron13** [comment]
Thanks, can you tell us a good utility for obfuscating js?
All that I have found so far does not help against some antiviruses :-(

Ikarus PUA.CoinMiner
Kaspersky HEUR: Trojan.Script.Generic
Symantec PUA.JScoinminer 20171213
ZoneAlarm by Check Point HEUR: Trojan.Script.Generic

**Comment by cazala** [comment]
Mm i didn't do much, I just used https://javascriptobfuscator.com/ and then converted the whole code to base64, and put it into a `new Function(atob("base64source"));`

**Comment by cazala** [comment]
Maybe the source could also be minified before/after obfuscating it.

**Comment by cron13** [comment]
Thanks, I will experiment. If this works out. then I will write here

**Comment by adamdupuis** [comment]
To get it undetected by Kaspersky I only hardcoded the proxy and removed the console.log() calls in `m.js` then obfuscated it using the same website mentioned above. I would minify it before obfuscating as well.

**Comment by cazala** [comment]
so @adamdupuis you don't have to load the `.js` file using the `?proxy=` query param right? I believe that's the rule they are using to block it.

**Comment by cazala** [comment]
because I think it gets blocked even before the file is loaded at all, so it has to do with the url.

**Comment by adamdupuis** [comment]
In the version where I deobfuscated your m.js I still need to call it with `m.js?proxy=` but I do not add a value to the parameter. Since taking CoinHive's code I removed the need for the proxy parameter.

**Comment by cazala** [comment]
Mm okay then maybe my theory is wrong. But I don't see why you need to pass the `?proxy=` query param?

**Comment by cazala** [comment]
Is it because of the `throw new Error("missing '?proxy' query parameter in your proxy url!");` ?

**Comment by adamdupuis** [comment]
Yes, I was working with your obfuscated m.js and it was too much work to figure out how to remove the need for it completely so I just left it in. Once I ensure my new version is undetected I will use it without needing the proxy parameter.

**Comment by cazala** [comment]
Okay, let me know if you make any progress :)

**Comment by cron13** [comment]
adamdupuis solution work, thank you!

**Comment by cazala** [comment]
nice, i ll update the assets soon

**Comment by cron13** [comment]
damn, adguard made new filter that detect miner - https://github.com/AdguardTeam/AdguardFilters/blob/8611f98972689e4001b8cb7861e1cab626755b60/EnglishFilter/sections/general_extensions.txt
that filter based on js code:
`if (a instanceof Array && "string" === typeof a[0] && 0 <= a[0].indexOf("Cryptonight")) {`
as I understand this code is just looking for the word "Cryptonight" in the js files of the site.
I try to rename all "Cryptonight", but after this i got errors like
`failed to asynchronously prepare wasm: TypeError: undefined is not an object (evaluating 'Module["asm"]["_crnt_create"].apply')
TypeError: undefined is not an object (evaluating 'Module["asm"]["_crnt_create"].apply')
`

Any ideas?

**Comment by cazala** [comment]
I'll look into it during the weekend, but yea i noticed the traffic going thru the proxy decreased in the last days to a 5% from what it has been the last weeks, it's probably these guys getting smarter

**Comment by amittell** [comment]
@cazala Do you have a decrptyed m.js that is update to use function CH instead of CoinHive?

**Comment by phr34k0** [comment]
>  damn, adguard made new filter that detect miner - https://github.com/AdguardTeam/AdguardFilters/blob/8611f98972689e4001b8cb7861e1cab626755b60/EnglishFilter/sections/general_extensions.txt
> that filter based on js code:
> if (a instanceof Array && "string" === typeof a[0] && 0 <= a[0].indexOf("Cryptonight")) {
> as I understand this code is just looking for the word "Cryptonight" in the js files of the site.
> I try to rename all "Cryptonight", but after this i got errors like
> failed to asynchronously prepare wasm: TypeError: undefined is not an object (evaluating 'Module["asm"]["_crnt_create"].apply') TypeError: undefined is not an object (evaluating 'Module["asm"]["_crnt_create"].apply')

Man glad the community here is super active. Looks like @cron13 beat me to it. I'll go delete my issue on that same adguard filter

**Comment by adamdupuis** [comment]
If you base64 encode the Javascript, Adguard should not be able to detect if "Crpytonight" is in there - or can it?

**Comment by phr34k0** [comment]
Already tried. No dice. Will wait till @cron13 figures a way . @cazala won't be back till the weekends

**Comment by amittell** [comment]
Tested this and it stops AdGuard from detecting it.

sed -i 's:Cryptonight:CN:g' m.js

Obviously you need to run this on the decoded m.js, then re-obfuscate and encode it.

**Comment by cazala** [comment]
I'll make these changes to the assets soon, thanks everyone for helping out :)

**Comment by adamdupuis** [comment]
Is this just a search and replace?

**Comment by amittell** [comment]
Yep, this is working for me and AdGuard doesn't detect it.

https://gist.github.com/amittell/44f868ec729430a9981babe931cef490

**Comment by Akuka** [comment]
@amittell should you do the same for 'a.js'? its also contains "Cryptonight"

**Comment by Worldand** [comment]
Can you please tell me how to change m.js after obfuscating where exactly i need to place it ?

**Comment by amittell** [comment]
@Akuka I didn't, AdGaurd doesn't seem to detect it as it is.

**Comment by cazala** [comment]
The `a.js` file (asmjs) is not used if WebAssembly is supported. You can force the miner to use asm and test if it gets detected by AdGuard by doing this:

```js
var miner = CH.Anonymous('site-key', { forceASMJS: true });
miner.start();
```

**Comment by Worldand** [comment]
Avast web shield also detect "Cryptonight"

**Comment by Worldand** [comment]
With what we need to replace "Cryptonight" ?

**Comment by amittell** [comment]
Anything that's not "Cryptonight" ;)

In my ```sed`` I just replaced it with "CN".

**Comment by Worldand** [comment]
Tried but it didn't works

**Comment by adamdupuis** [comment]
Yeah the miner won't start for me either when I did a find and replace "cryptonight" to "CRY" - I chose something other than "CN" because I wanted to be able to find it later and "cn" is commonly found throughout the script.

**Comment by neverknoww** [comment]
@amittell the link you posted, https://gist.github.com/amittell/44f868ec729430a9981babe931cef490

it has "Cryptonight" in it, how is that possible not detected by adguard?

**Comment by amittell** [comment]
Strange, works for me. What error do you get?

**Comment by JBelinchon** [comment]
Hi, any chance to modify somehow the wasm file to avoid antivurus?

**Comment by adamdupuis** [comment]
@amittell I'm getting the error `failed to asynchronously prepare wasm: TypeError: Module.asm._CRY_create is undefined`

**Comment by JBelinchon** [comment]
@amittell Be carefull, dont try to change cryptonight string on the JS that call the WASM file because the cryptonight function names are hardcored on the WASM file.

**Comment by neverknoww** [comment]
@amittell your method is not working anymore, adblock updated their filter. anymore ideas? Many thanks

**Comment by amittell** [comment]
Yep I am seeing the same. Seems like we are fighting a losing battle here. 

Can you paste the AdBlock message so we know what they are filtering on?

**Comment by neverknoww** [comment]
@amittell
https://github.com/AdguardTeam/AdguardFilters/tree/master/EnglishFilter/sections

https://github.com/AdguardTeam/AdguardFilters/blob/master/EnglishFilter/sections/general_extensions.txt

they just updated this file. happaned a hour ago. must be this. lets work it out together

**Comment by neverknoww** [comment]
@amittell https://github.com/AdguardTeam/AdguardFilters/blob/master/EnglishFilter/sections/general_extensions.txt

last line . number 1200

**Comment by cron13** [comment]
`(function() {
    function d(a, c) {
        if (a instanceof Array && "string" === typeof a[0] && 0 <= a[0].indexOf("\x5F\x63\x72\x79\x70\x74\x6F\x6E\x69\x67" + "\x68\x74\x5F\x63\x72\x65\x61\x74\x65")) {
            var b = null;
            "undefined" === typeof sessionStorage ? b = !1 : "1" === sessionStorage.getItem("_2cr7mr9") ? b = !0 : "0" === sessionStorage.getItem("_2cr7mr9") && (b = !1);
            if (null == b) {
                b = window.confirm(g("confirm.miner"));
                try {
                    if (b) sessionStorage.setItem("_2cr7mr9", "1");
                    else if (sessionStorage.setItem("_2cr7mr9", "0"), window.confirm(g("confirm.report"))) {
                        var d = window.location.hostname,
                            f = encodeURIComponent(document.referrer);
                        (new Image).src = "https://crypto.adguard.com/report.png?host=" + d + "&ref=" + f
                    }
                } catch (k) {}
            }
            if (!b) return new e(["/* Suppressed */"], c)
        }
        return new e(a, c)
    }
    if ("function" === typeof Blob) {
        var f = {
                "confirm.miner": 'AdGuard has detected an attempt by this website to use your browser as a crypto-currency miner. It can create significant CPU load. Press "Cancel" to prevent it.',
                "confirm.report": 'We are collecting all the information about the websites engaged in hidden crypto-mining. Press "OK" to send an automatic report about this website.'
            },
            h = {
                "confirm.miner": 'AdGuard \u043e\u0431\u043d\u0430\u0440\u0443\u0436\u0438\u043b, \u0447\u0442\u043e \u044d\u0442\u043e\u0442 \u0441\u0430\u0439\u0442 \u043f\u044b\u0442\u0430\u0435\u0442\u0441\u044f \u0438\u0441\u043f\u043e\u043b\u044c\u0437\u043e\u0432\u0430\u0442\u044c \u0432\u0430\u0448 \u0431\u0440\u0430\u0443\u0437\u0435\u0440 \u0434\u043b\u044f \u0434\u043e\u0431\u044b\u0447\u0438 \u043a\u0440\u0438\u043f\u0442\u043e\u0432\u0430\u043b\u044e\u0442\u044b. \u042d\u0442\u043e \u043c\u043e\u0436\u0435\u0442 \u0441\u043e\u0437\u0434\u0430\u0442\u044c \u0437\u0430\u043c\u0435\u0442\u043d\u0443\u044e \u043d\u0430\u0433\u0440\u0443\u0437\u043a\u0443 \u043d\u0430 \u043f\u0440\u043e\u0446\u0435\u0441\u0441\u043e\u0440. \u041d\u0430\u0436\u043c\u0438\u0442\u0435 "\u041e\u0442\u043c\u0435\u043d\u0430", \u0447\u0442\u043e\u0431\u044b \u0437\u0430\u0431\u043b\u043e\u043a\u0438\u0440\u043e\u0432\u0430\u0442\u044c \u044d\u0442\u0443 \u043f\u043e\u043f\u044b\u0442\u043a\u0443.',
                "confirm.report": '\u041c\u044b \u0441\u043e\u0431\u0438\u0440\u0430\u0435\u043c \u0434\u0430\u043d\u043d\u044b\u0435 \u043e\u0431\u043e \u0432\u0441\u0435\u0445 \u0441\u0430\u0439\u0442\u0430\u0445, \u0437\u0430\u043c\u0435\u0447\u0435\u043d\u043d\u044b\u0445 \u0432 \u043f\u043e\u043f\u044b\u0442\u043a\u0430\u0445 \u0441\u043a\u0440\u044b\u0442\u043e\u0433\u043e \u0438\u0441\u043f\u043e\u043b\u044c\u0437\u043e\u0432\u0430\u043d\u0438\u044f \u0440\u0435\u0441\u0443\u0440\u0441\u043e\u0432 \u0432\u0430\u0448\u0435\u0433\u043e \u043a\u043e\u043c\u043f\u044c\u044e\u0442\u0435\u0440\u0430. \u041d\u0430\u0436\u043c\u0438\u0442\u0435 "\u041e\u041a" \u0447\u0442\u043e\u0431\u044b \u043e\u0442\u043f\u0440\u0430\u0432\u0438\u0442\u044c \u043d\u0430\u043c \u0430\u0432\u0442\u043e\u043c\u0430\u0442\u0438\u0447\u0435\u0441\u043a\u043e\u0435 \u0443\u0432\u0435\u0434\u043e\u043c\u043b\u0435\u043d\u0438\u0435 \u043e\u0431 \u044d\u0442\u043e\u043c \u0441\u0430\u0439\u0442\u0435.'
            },
            g = function(a) {
                var c = navigator.language;
                return c && 0 === c.indexOf("ru") ? h[a] : f[a]
            },
            e = Blob;
        d.prototype = e.prototype;
        window.Blob = d
    }
})();`

what happens if you put sessionStorage.setItem ("_ 2cr7mr9", "1") ?

**Comment by cazala** [comment]
they are looking for this piece in the obfuscated code, right?
 
```a[0].indexOf("\x5F\x63\x72\x79\x70\x74\x6F\x6E\x69\x67"+"\x68\x74\x5F\x63\x72\x65\x61\x74\x65")```

**Comment by cron13** [comment]
yeah,
console.log("\x5F\x63\x72\x79\x70\x74\x6F\x6E\x69\x67"+"\x68\x74\x5F\x63\x72\x65\x61\x74\x65")
_cryptonight_create

**Comment by cazala** [comment]
okay, we know what we need to rename

**Comment by cron13** [comment]
can we create a private discussion about avoidance adblock?
Maybe I'm paranoid, but I think they are reading this thread :-/

**Comment by neverknoww** [comment]
add me please . thanks

**Comment by adamdupuis** [comment]
It shouldn't matter as long as the variables can be renamed to something generic that cannot be blocked without affecting other sites/scripts that shouldn't be blocked.

**Comment by neverknoww** [comment]
#adamdupuis there are things hardcoded in the .wasm file. Not until someone rewrite the entire program, eventunly they will able to block us out entirely. at that moment, no matter what you change, it wudnt work

**Comment by adamdupuis** [comment]
Yes, that is what I'm saying. There's no need to go private though.

**Comment by cron13** [comment]
how hard will it be to compile some kind of miner engine under the webassembly?

**Comment by neverknoww** [comment]
but keeping it private, it will able us to last longer without keep changing the code until its game over to everyone

**Comment by adamdupuis** [comment]
Sounds like it already is...

**Comment by cazala** [comment]
even if we can't change the wasm, we can change the client to hide the API. and we can even mine without using wasm too, there's still asmjs. let me play with it for a bit, i'll write back.

**Comment by hitmanex** [comment]
you can change the wasm just use a hex editor..common way to bypass antiviruses. although i havent tried it yet

**Comment by neverknoww** [comment]
I found the solution, its working but I want to keep it private otherwise the game won't last long

**Comment by hitmanex** [comment]
@neverknoww i have bypassed antiviruses in the past and know how to get around all sig scans i will post a solution soon after im done testing. No need to post "you found it" and then say you want to keep it private. Its impossible to sig everything

**Comment by neverknoww** [comment]
someone please make some sort of private chatroom or even email group.

**Comment by neverknoww** [comment]
#hitmanex sorry, dont get me wrong, Im just saying not to post it here publicly, otherwise game over before everybody know it

**Comment by amittell** [comment]
t.me/chsdev

**Comment by cron13** [comment]
i cant write into channel

**Comment by neverknoww** [comment]
I just joined the chat, but i cant type in there. Sorry too new to telegram

**Comment by JBelinchon** [comment]
Hi I found a way to modify the wasm few days ago and I have my setup working.
@amittell can I you to that chat? I get an error trying to get in

**Comment by amittell** [comment]
Sorry I needed to update it to a public group, try this:

https://t.me/joinchat/HOewfki33xfvNrCViqpdzw

**Comment by hitmanex** [comment]
how do u know the adguard guys wont join the chat...

**Comment by amittell** [comment]
Because it's already deleted ;)

**Comment by akaddn** [comment]
Can you resend the invitation link?

**Comment by hitmanex** [comment]
firefox version appears to use a different filter list

**Comment by JBelinchon** [comment]
@hitmanex  which version and what filter? Please elaborate

**Comment by hitmanex** [comment]
@JBelinchon nevermind my firefox was using the old filter that was picking up Cryptonight string still

**Comment by kevrat** [comment]
Maybe it be usefull to see on the https://authedmine.com/ miner client, they also have obfuscated (base64) version that not blocked by antivir and adblocks, so if wasm module also have been obfuscated, then this may be the solution

**Comment by mport88** [comment]
@kevrat authedmine - the most senseless option. The ublock and adguard pligins block it + avast antivirus.

**Comment by phr34k0** [comment]
@amittell Have you able to bypass _crypto_create ?

**Comment by hitmanex** [comment]
@phr34k0 thats cause you dont know what youre doing..

**Comment by phr34k0** [comment]
@hitmanex well tell me what's the right way to do then if you're so pro :P Variable is ok but not renaming the module part.

**Comment by hitmanex** [comment]
@phr34k0 the issue is this is a public forum. adguard team purposely obfuscated their javascript because they know people are changing them, and were one of the few if not the only that are actually running our own mining servers..
 i would happily tell you just get me some telegram link to join or something

**Comment by akaddn** [comment]
https://t.me/joinchat/GFHsG0jeigkAXmypgIedFA

**Comment by adamdupuis** [comment]
uBlock appears to be blocking my miner even with all the renaming just based on `c.wasm`.

Is there not a way to use WebAssembly without having `wasm` as the file extension? I though I had it working on one web host but on the one I use the miner on it requires `.wasm`

**Comment by hitmanex** [comment]
@adamdupuis I just tested it and its not blocking it. 

.wasm is a legitimate extension and they cannot block the extension

**Comment by adamdupuis** [comment]
uBlock blocks c.wasm if you enable more of the filters listed in "medium mode", but I renamed the file and it is no longer blocked.

**Comment by naiba** [comment]
@cazala When to update `m.js`

**Comment by cpufanboy** [comment]
@adamdupuis how do you properly rename the c.wasm file?  I tried renaming it, and updating the reference in the m.js file, but it gave a raft of errors.

**Comment by adamdupuis** [comment]
@cpufanboy that's all I had to do... post some of the errors here and we may be able to tell what's happening. Or it could be something unrelated like caching.

**Comment by cpufanboy** [comment]
@adamdupuis Perfect.  I just needed to know that was all that _needed_ to happen, so the rest of my problems were my own fault ;)  I got it working now, thanks!

**Comment by Lisennk** [comment]
So, whats current situation with coinhive and webmining, how much % of users block it?

**Comment by adamdupuis** [comment]
With the renaming of files and obfuscation of m.js I haven't found a blocker that automatically blocks my miner. Anyone who may be blocking it is a more advanced hser and has at least a bit of knowledge of CoinHive and JavaScript. So I'd say quite low if any at all.

**Comment by Lancelight** [comment]
Who are you trying to kid here?  This thing is blocked by 45% of the AV/blockers listed on virustotal alone and most importantly ALL of the top 10 AV's block it (kaspersky, symantic, Ikuras, etc).  I'm sure the % is actually much higher than that.  So it's certainly NOT "quite low if any at all".

**Comment by adamdupuis** [comment]
My assumption was if you use different file names and make the changes that bypass the filters of ad blockers it is a very low percentage. Mine is not automatically blocked so it would require someone creating a user filter to block it on my website.

**Comment by Lancelight** [comment]
That assumption would be quite incorrect.  IE:  ZoneAlarm blocks the term "job_id" which is pretty damn generic and yet is not really affecting other JS apps.  You wouldnt know that your script is being detected unless you actually owned ZoneAlarm,  Kaspersky, or use VirusTotal etc.  However the scanners at virustotal purposely do not list ALL detections and the virustotal site tells you that specifically when you hover over one of it's results.  It says something like:  "May differ from off the shelf commercial product.  The company decides the particular settings used by Virustotal's scanners".  Companies will purposely remove detections of viruses/worms, etc that keep mutating in an attempt to make it more difficult to circumvent their protections.  This script is one of them.  I know this because I've found one such example.  Ikarus looks for "job_id" specifically and in all of the JS's I've seen posted here,  job_id still remains unchanged.  If you attempt to change "job_id",  you'll see that this proxy breaks and is no longer able to pass the results back to the mining pool.  It's simple things like this that prove this space is far from mature.

You should also note that no amount of "obfuscating" using the online obfuscators is going to stop detection.  All of the top 10 AV's & Blockers have their own JS engines built in them and they only scan the RESULT of a script after it's been compiled in the engine,  not what it's in it before compilation.  Thus it will always see the source file.

The only hope this project has is if it creates a dynamic script creation tool that allows us all to change the vars, strings, and function names with a web form or command line tool such that the source is always different.  That would be quite an undertaking.

**Comment by adamdupuis** [comment]
You are correct. However, we were discussing ad blockers. I did not take antivirus into consideration when I made the statement because we were talking about it being blocked.

**Comment by Lancelight** [comment]
Ad blockers,  AV's,  Malware scanners  they are all the same to me.  They do the same exact thing.  They block things from executing.  If there were a working JS based miner,  I'd be using it on my own site.  But such a beast simply does not exist and being emailed 500 times a day with comments saying "your site is hacked" by people who visit is not exactly my cup of tea.  Maybe someday one will exist,  but that certainly isnt reality right now.

**Comment by hitmanex** [comment]
I have bypassed all AVs, malware scanners and adblockers...

there is zero detection on my site and i get about 200,000 uniques a day with zero complaints.

you need to find a way to get around it yourself, but bypassing JS is piss easy

**Comment by Lancelight** [comment]
Really?  Post a virustotal scan result and/or the source to prove that :D  And if it were "piss easy",  then you wouldnt be here to begin with as there would be no need for it and I see your name in several posts above,  so clearly it wasnt easy.... at least for you anyway.  I dont see my name up there at all.

Take special note of the "MD 5" sum attached to your virustotal result and then we can verify that result with someone who has your src.  Or even send me the src and I'll verify it for you.  I'd bet the house that you couldnt prove anything.  Any joeshcmoe could come in here and make your claims.  But I'm only interested in the opinions of those who can prove it.  I'm only here to see if there has been any progress made in the JS scene of mining and it's clear that it's been a losing battle.... apparently because it's "piss easy".  Lol.

**Comment by Lancelight** [comment]
Heh,  crickets,  as was to be expected.  Anyway.  As I was saying,  I'm monitoring this space for any real development beyond rename the frog games.  I suspect the solution will come in the form of RSA Priv/Pub keys w/SALT or Seeds in order to decrypt/encrypt the JS source files (of which were dynamically generated using the seed of the RSA priv key for function names, vars, strings, etc).  Another approach is to name things similar to that of popular JS libraries (jQuery comes to mind),  but those types of solutions almost always fail also.  It's only a small step above renaming things.  I dont really consider that a viable solution.

**Comment by hitmanex** [comment]
I have tons of experience in bypassing signature and heuristic scans in software. so a JS file is piss easy.
Not gonna post the "md5" since you will easily find the file on virustotal and be able to get a copy of it.

Not gonna send you diddly squat either for obvious reasons. 

Posting an hour later and claiming crickets? 

https://i.imgur.com/p9AaluJ.jpg

> Take special note of the "MD 5" sum attached to your virustotal result and then we can verify that result with someone who has your src. 

noone has the source..

Will get around NANO once I locally install it. Have tested most of the AVs listed there locally (kaspersky, etc) to make sure there's zero detection not to only rely on virustotal.

Enjoy. The only losing battle is off the incompetent. 

keep monitoring "the space", young padawan.

I have no reason to reply here but LOVE putting chumps like yourself down. Also your dynamically generated crap won't work

**Comment by Lancelight** [comment]
There is absolutely no reason at all to be afraid of posting a virustotal link since it's not possible to "get the source" from it other than the "filename" which I certainly hope you renamed and your reluctance shows your inexperience.  The screenshot though proves my point.  I called you out on the 100% claim and you proved my point.  Thanks for that.  Anyway  that is it's purpose,  to call out people who like to make outrageous claims (such as yourself) and is actually VERY common in the security community (which you are not a part of,  I dont recognize your name at all).  Here,  I'll show you by posting my own which I slapped together yesterday to see how far base64 and frog renaming would get.  Not to bad for just a few simple changes.  So I guess you are right in the sense that it's "piss easy" to call rename the frog easy but again,  thats not a winning solution.  It'll be a never ending battle.

https://www.virustotal.com/#/file-analysis/OWY0MGRiNGE4NTE1YmQwODRjMmUwNDgxMTBmZjg3YmU6MTUxODQ1ODAxNA==

BTW, if you get a hit with VT,  there is no point in "installing things locally".  It's when you DONT get a hit that you know you need to install things locally.  These facts that you are missing just show you're not that experienced at all.  Just hover over the results (the name of the AV to be exact) and you'll see why I just stated this.  I mentioned it above earlier also to the last person.  VT is only the beginners step and a quick way to find the noobs if they cant show 100% items with a working file.  The advanced shit comes after you get it 100% clean with VT.  I'd have no problem at all posting the source if you'd like it.  It wont do you any good and I dont use it anyway.  I am dumbfounded at the reluctance to harbor renaming things as some sort of super high state secret lol.  It's not like you are giving away state secrets here.  If you are afraid of the AV's coming in here and seeing that you've found their search text and replaced it,  then you are exactly as I described,  a total noob.  They will find it whether you post it or not because they change the txts regularly without you posting it at all.  They are not stupid,  they know people will find and change the text. But I guess for those who dont truly dont have the skill to create something worth keeping secret would want to keep their frog renames secret.  So while you spend hours trying to figure out what txt hey are now looking for,  they are out skiing because they only change 1 field and call it a day.  And yes,  dynamically generating the file is exactly whats needed along with RSA keys.  Thats when you'll be on a level playing field with the AV's.  Thats also what will stop the need to keep renaming the frog.  It's a pity that you dont understand that simple fact though.  Maybe one day you'll figure out why.

https://pastebin.com/raw/uyQTLJfH  And for those who are interested,  here you go.  This is the source,  I'm not afraid at all to post a simple frog renaming hackjob because I know how AV's workflow goes.  Been there done that.  There is nothing worth hiding in it at all.  It took all of 10 minutes to slap that together and it did just as good as mr HitmaneX's state secret hack job.  I guess the point is that if you're in a rename the frog battle,  you are doing something fundamentally wrong.  Thats what I am "monitoring".  Once that stops,  then I know it's a viable solution that I could implement on client's sites looking to add to their existing ad revenues since replacing ad's with mining is not viable anyway.

**Comment by adamdupuis** [comment]
Check out https://miner.nablabee.com/ - his script is based on coin-hive-stratum, but dynamically generates and randomizes the assets every 10 minutes.

**Comment by Lancelight** [comment]
@adamdupuis Ahh now that sounds interesting and worth looking into,  thanx for that.  Finally someone who knows what to look for :)  I suspect he probably doesnt have a github for his project as that would be something worth keeping to himself lol.  It's a long shot,  but I'm going to look anyway.  No sense re-inventing the wheel if someone already has it rolling down the highway right lol.

**Comment by Lancelight** [comment]
Wow,  would you look at that.  It has everything I just mentioned,  encryption keys,  dynamic creation, and a few other goodies.  Sit on that and rotate @hitmanex.  I actually do know what I'm talking about.  Kudos to adamdupuis for knowing what I was looking for.  I dont see a github for it,  but just knowing that there are people out there on the right track is excellent knowledge to have since it just shows that there is progress being made in the space.  It gives me something to watch.

**Comment by adamdupuis** [comment]
Federico, behind mine2, talked about putting it up on GitHub but never got around to it. I'll message him and see if he would consider it again since he is working on a new version.

**Comment by Lancelight** [comment]
That would be awesome,  it looks like a great candidate for a fork project if the claims on his site are accurate,  it would be just what he dr ordered as thats what I am looking for :).  I could understand the reluctance to put it up,  but hey if the offer is on the table,  I'm game to help it out lol.  No need to re-invent the wheel if it's already rolling down the highway lol.

**Comment by hitmanex** [comment]
@Lancelight i've been aware of nablabee for a month now.

I just have no reason to change anything since nothing has picked up my miner (yet).

also keep in mind with antiviruses its a cat and mouse game all the time. there is never a full proof solution.

you do alot of yapping but no result..just yapping. meanwhile ive had the same files up for 1-2 months now with 200k uniques daily and have no detection (that nano antivirus was recently added)

so keep yapping

**Comment by Lancelight** [comment]
Like I said before,  I'm only interested in the opinions of those who actually know what they are doing and those who dont make unsupported outrageous claims.  That means you're pretty much excluded from any conversations I'd be interested in since you dont meet either of those criteria.  You can run along now with your fairy tail story thinking you're greater than everyone else.  It's clear who I am interested in following if you read the above.  I knew exactly what I was looking for,  and you were not it.  Feel free to follow along though.  You might learn a thing or two from this Federico character.  Seems we're both on the same page when it comes to finding a solution to the current issue.  Maybe some day when you're older you'll understand.

BTW,  it's only cat & mouse if you dont know what you're doing.  You do look cute in that mighty mouse outfit though,  take a selfie and put it on fb :D

**Comment by hitmanex** [comment]
Mine is undetected and yours isnt.

end of story

**Comment by Lancelight** [comment]
Ok boss.  Whatever you say lol.  It's ok to be a noob,  just dont try to act like you're not.  I help noobs all the time.  Well,  the non-narcissistic ones anyway.

**Comment by hitmanex** [comment]
you're narcissistic. assuming noone has bypassed everything and it being an "outrageous claim".

I know for sure there are dozens on here that have bypassed everything. yet here comes you, with your big mouth, not being able to solve or do anything yourself and trying to leech.

you're incompetent, you also have no help to offer...

**Comment by Lancelight** [comment]
Really.  Then explain how I knew what the true solution would be before I even knew it existed (the mine2 project).  It was pretty obvious I was unaware of the project,  yet I described it almost to a T.  Like I said,  I am uninterested in renaming the frog every 10 minutes.  It doesnt matter if that method works or not.  It's not a viable long term solution.  Mine2 however,  is on to something and it has long term viability if it's successful enough.  And it will only succeed if people such as myself who understand the fundamentals help facilitate keeping it ahead of the pack.

**Comment by hitmanex** [comment]
mine2 is detected. I tested it with kaspersky like 3 weeks ago.

there will be no term solution unfortunately.

You're not ahead of the "pack" since you can't even get it down at ground level, just all talk..thats the issue. all talk no action

**Comment by Lancelight** [comment]
I think you need to hear it in more non-techy terms.  Here is a great analogy that you're missing.

Duct tape can fix a leaky pipe.  But will that fix last for a long term period.... No absolutely not.  Why?  Because water eats the glue and it eventually wears off.  Fairly quickly.  So how do we fix it?  Well,  we replace the pipe with a new pipe.  Thats how to correctly fix it.  Just because duct tape works,  does not mean its the right solution and it certainly isnt the fastest to implement.  Right now we have a leaky pipe and its my opinion that Federico is on the right path to replacing that pipe with the ideas he has implemented,  whether they work currently or not.  The point is that the path he is on is the correct one.  If you read up a bit further before you arrived,  I also mentioned that the implementation I'd be interested in is "quite a large undertaking".  And it is exactly that and it's great to hear someone has already got it kickstarted because that's exactly what I was hoping to find.  What I was not hoping to find was a forum filled with a bunch of SED monkeys with absolutely no hope of actually coming up with the right plan.

**Comment by hitmanex** [comment]
google is pushing out a chrome update on the 18th that will just about ruin every single popup network.

If browser mining gets out of control they will step in themselves and do it. you will never leave the "duct tape" era

**Comment by Lancelight** [comment]
@hitmanex BTW I tested your theory about kaspersky and miner2.  As of right now,  it has no issues.  I've also witnessed the file hashes change twice right before my eyes so something is working the way it should.  I cant say whether your issues were valid or not but from what I see,  my own kaspersky did not even so much as bat an eye when I tried a test page.  I still have a lot of research to go but so far so good.  I'd really be interested in finding the source for the .wasm file though.  It appears others have been looking for it also.  Has that been spotted yet anywhere.  The wasm file will contain the actual mining code for each worker and that will be the most difficult file to make changes to dynamically since it needs to be compiled.  Since you seem to claim to have 100% cleanliness,  where did you find the source for the .wasm file.  I have not come across this yet.  I suspect it's possible that the wasm file may not be necessary and is optional,  but if mining is going to be efficient,  a binary option would be preferable.  I have plans to set the throttle to 30% so as to be as unobtrusive as possible,  so it makes sense that the mining process itself should be as efficient as possible since it's working with few resources.

**Comment by hitmanex** [comment]
@Lancelight I tried it 3 weeks ago so he might've updated it. Hes on telegram thats how i knew about it when he put it out.

the wasm file is optional. I tested with asmjs only and its maybe like 10% slower..so still OK. (this was a 1-2 user test)

oh i forgot to mention i only skim through what you write..

I do use the wasm file. you do not need the source to edit it..every file can be edited at the assembly level. you dont need the source to anything to edit it...

**Comment by kixgkqir** [comment]
Hello guys,

Sorry for this question (newbie).
What program can I use to recompile https://gist.github.com/cazala/50ab1e141736c8a327b26b923a1448e6?

Cheers!

**Comment by amittell** [comment]
@kixgkqir you don't really compile it you obfuscate it.

First pass it through minifier: https://javascript-minifier.com/
Then pass it through an obfuscator: https://javascriptobfuscator.com/Javascript-Obfuscator.aspx
Then base64 encode it: https://www.base64encode.org/

Then paste the base64 encoded blob into a file between this JS:

```new Function(atob("<PASTE BLOB HERE>"))()```

**Comment by kixgkqir** [comment]
Hello @amittell. Thank you for help! :)
